docs/UserGuide.md

```diff
@@ -181,6 +144,92 @@ Expected output (fail):
+
+Format: `view INDEX`
+
+Parameters:
+* `INDEX`: Index of a fosterer to view their details shown in the list obtained by the find/list command
+
+<div markdown="span" class="alert alert-primary">
+  :exclamation: <b>Important:</b>
+Some commands are not available while on profile view page. 
+The list of available commands are: 
+<ul>
+  <li> <code>help</code> </li>
+  <li> <code>edit</code> </li>
+  <li> <code>exit</code> command to <b>exit the profile view page</b> </li>
+</ul>
+</div> 
+
+Example:
+* `list` followed by `view 2` to view the profile of the 2nd fosterer in the address book.
+
+Expected output (success):
+```agsl
+Viewing the profile of a fosterer...
+```
+Expected output (fail):
+```agsl
+Oops! Invalid fosterer index provided, please check again.
+```
+### Editing a fosterer's detail : `edit`
+
+Edits or views the details of a fosterer stored in the address book.
+
+Format 1: `edit INDEX [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS] [housing/HOUSING_TYPE] [availability/AVAILABILITY] [animal/ANIMAL_NAME] [animalType/TYPE_OF_ANIMAL_FOSTERED/WANTED] [t/TAG…]`
+
+Format 2: `edit INDEX`
+
+Parameters:
+* `INDEX`: Index of a fosterer to be edited shown in the list obtained by the find/list command
+* `NAME`: New name of the fosterer 
+* `PHONE_NUMBER`: New phone number of the fosterer
+* `EMAIL`: New email of the fosterer
+* `ADDRESS`: New address of the foster family
+* `HOUSING_TYPE` (case-sensitive) : HDB / Condo / Landed / nil
+* `AVAILABILITY` (case-sensitive) : Available / NotAvailable / nil
+* `ANIMAL_NAME`
+  * `ANIMAL_NAME` (if `availability/NotAvailable`) : New name of animal fostered
+  * `ANIMAL_NAME` (if `availability/nil`) : nil
+  * `ANIMAL_NAME` (if `availability/Available`) : nil
+* `TYPE_OF_ANIMAL_FOSTERED/WANTED` (case-sensitive) :
+  * `TYPE_OF_ANIMAL_FOSTERED` (if `availability/NotAvailable`) : current.Dog / current.Cat / nil
+  * `TYPE_OF_ANIMAL_WANTED` (if `availability/Available`) : able.Dog / able.Cat / nil
+  * `TYPE_OF_ANIMAL_FOSTERED/WANTED` (if `availability/nil`) : nil
+
+<div markdown="span" class="alert alert-primary">
+  :bulb: <b>Tip:</b>
+The index of the fosterer has to be provided, however the number of parameters to be edited can vary from zero to all fields.
+</div> <br>
+<div markdown="span" class="alert alert-primary">
+  :exclamation: <b>Important:</b>
+If the parameters are not provided, <b><code>edit INDEX</code> operates the same way as <code>view INDEX</code></b>, leading you to the profile page of the person at index <code>INDEX</code> in the addressbook. 
+</div> 
+<br>
+
+Examples:
+*  `list` followed by `edit 3 n/John` edits the name of the 3rd fosterer in the address book to John.
+*  `list` followed by `edit 1 p/12345678 animal/Bob` edits the phone number and the pet name of the 1st fosterer in the address book to 12345678 and Bob respectively.
+*  `list` followed by `edit 2` changes the view to the profile page of the 2nd fosterer in the address book.
```

Comment:
maybe can add "since parameters are not provided" at the end of the sentence to better link back to what you said earlier in line 209